addons/binding/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/handler/NeeoRoomHandler.java

```diff
@@ -0,0 +1,292 @@
+     * @param channelSection the non-empty channel section
+     * @param channelKey the non-empty channel key
+     * @param channelId the non-empty channel id
+     */
+    private void refreshChannel(NeeoRoomProtocol protocol, String channelSection, String channelKey, String channelId) {
+        Objects.requireNonNull(protocol, "protocol cannot be null");
+        NeeoUtil.requireNotEmpty(channelSection, "channelSection must not be empty");
+        NeeoUtil.requireNotEmpty(channelId, "channelId must not be empty");
+
+        switch (channelSection) {
+            case NeeoBindingConstants.CHANNEL_GROUP_STATEID:
+                switch (channelId) {
+                    case NeeoBindingConstants.CHANNEL_ACTIVESCENARIOS:
+                        protocol.refreshActiveScenarios();
+                        break;
+                    case NeeoBindingConstants.CHANNEL_CURRENTSTEP:
+                        protocol.refreshCurrentStep();
+                        break;
+                }
+                break;
+            case NeeoBindingConstants.CHANNEL_GROUP_RECIPEID:
+                NeeoUtil.requireNotEmpty(channelKey, "channelKey must not be empty");
+                switch (channelId) {
+                    case NeeoBindingConstants.CHANNEL_NAME:
+                        protocol.refreshRecipeName(channelKey);
+                        break;
+                    case NeeoBindingConstants.CHANNEL_TYPE:
+                        protocol.refreshRecipeType(channelKey);
+                        break;
+                    case NeeoBindingConstants.CHANNEL_ENABLED:
+                        protocol.refreshRecipeEnabled(channelKey);
+                        break;
+                    case NeeoBindingConstants.CHANNEL_STATUS:
+                        protocol.refreshRecipeStatus(channelKey);
+                        break;
+                }
+                break;
+            case NeeoBindingConstants.CHANNEL_GROUP_SCENARIOID:
+                NeeoUtil.requireNotEmpty(channelKey, "channelKey must not be empty");
+                switch (channelId) {
+                    case NeeoBindingConstants.CHANNEL_NAME:
+                        protocol.refreshScenarioName(channelKey);
+                        break;
+                    case NeeoBindingConstants.CHANNEL_CONFIGURED:
+                        protocol.refreshScenarioConfigured(channelKey);
+                        break;
+                    case NeeoBindingConstants.CHANNEL_STATUS:
+                        protocol.refreshScenarioStatus(channelKey);
+                        break;
+                }
+                break;
+        }
+    }
+
+    /**
+     * Simply cancels any existing initialization tasks and schedules a new task
+     *
+     * @see org.eclipse.smarthome.core.thing.binding.BaseThingHandler#initialize()
+     */
+    @Override
+    public void initialize() {
+        NeeoUtil.cancel(initializationTask.getAndSet(scheduler.schedule(new Runnable() {
```

Comment:
This is simple protectionist programming against bugs in the platform calling initialization twice.  This actually did occur with my russound addon in the 2.0 [maybe 2.1] stage - was a complete pain to track down.  Since it really doesn't hurt performance and is fairly easy to read - I do this in all my plugins now to protect against another bug cropping up in the platform